### PR TITLE
Spring cleaning: deprecate -> 4.2

### PR DESCRIPTION
**Pull request by dellaert**
- removed Lie{Scalar|Vector|Matrix} files completely
- Changed flag to V42
- moved a lot of deprecated things inside ifdef blocks
- made everything compile without deprecated methods
- removed deprecated methods from wrapper
- I do allow AHRSFactor deprecated methods: too much work to fix

### CODE REVIEW

File: gtsam/3rdparty/Eigen/Eigen/src/Core/DenseBase.h [hunk 1]
```diff
@@ -298,15 +298,15 @@ template<typename Derived> class DenseBase
 
     /** \internal
       * Copies \a other into *this without evaluating other. \returns a reference to *this.
-      * \deprecated */
+      * @deprecated */
```
Comment (ProfFan) [inline gtsam/3rdparty/Eigen/Eigen/src/Core/DenseBase.h:301]: same

File: gtsam/3rdparty/Eigen/Eigen/src/Core/TriangularMatrix.h [hunk 2]
```diff
@@ -435,12 +435,12 @@ template<typename _MatrixType, unsigned int _Mode> class TriangularViewImpl<_Mat
...
     { return *this = other.derived().nestedExpression(); }
 
-    /** \deprecated */
+    /** @deprecated */
     template<typename OtherDerived>
     EIGEN_DEVICE_FUNC
     void lazyAssign(const TriangularBase<OtherDerived>& other);
 
-    /** \deprecated */
+    /** @deprecated */
```
Comment (ProfFan) [inline gtsam/3rdparty/Eigen/Eigen/src/Core/TriangularMatrix.h:443]: same

File: gtsam/geometry/Cal3.h [hunk 3]
```diff
@@ -170,7 +170,7 @@ class GTSAM_EXPORT Cal3 {
     return K;
   }
 
-#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V41
+#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V42
   /** @deprecated The following function has been deprecated, use K above */
   Matrix3 matrix() const { return K(); }
```
Comment (ProfFan) [inline gtsam/geometry/Cal3.h:175]: `GTSAM_DEPRECATED`

File: gtsam/geometry/Cal3Bundler.h [hunk 4]
```diff
@@ -97,7 +97,7 @@ class GTSAM_EXPORT Cal3Bundler : public Cal3 {
 
   Vector3 vector() const;
 
-#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V41
+#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V42
   /// get parameter u0
   inline double u0() const { return u0_; }
```
Comment (ProfFan) [inline gtsam/geometry/Cal3Bundler.h:102]: `GTSAM_DEPRECATED`

File: gtsam/geometry/SimpleCamera.cpp [hunk 5]
```diff
@@ -21,7 +21,7 @@
 
 namespace gtsam {
 
-#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V41
+#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V42
   SimpleCamera simpleCamera(const Matrix34& P) {
```
Comment (ProfFan) [inline gtsam/geometry/SimpleCamera.cpp:25]: `GTSAM_DEPRECATED`

File: gtsam/linear/GaussianConditional.cpp [hunk 6]
```diff
@@ -193,12 +193,14 @@ namespace gtsam {
   }
 
   /* ************************************************************************* */
+#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V42
   void GaussianConditional::scaleFrontalsBySigma(VectorValues& gy) const {
```
Comment (ProfFan) [inline gtsam/linear/GaussianConditional.cpp:197]: `GTSAM_DEPRECATED`

File: gtsam/navigation/AHRSFactor.h [hunk 7]
```diff
@@ -182,24 +181,26 @@ class GTSAM_EXPORT AHRSFactor: public NoiseModelFactor3<Rot3, Rot3, Vector3> {
...
+  /// @deprecated static function, but used in tests.
+  static Rot3 predict(
       const Rot3& rot_i, const Vector3& bias,
-      const PreintegratedAhrsMeasurements preintegratedMeasurements);
+      const PreintegratedAhrsMeasurements& pim, const Vector3& omegaCoriolis,
+      const boost::optional<Pose3>& body_P_sensor = boost::none);
 
+#ifdef GTSAM_ALLOW_DEPRECATED_SINCE_V42
   /// @deprecated name
   typedef PreintegratedAhrsMeasurements PreintegratedMeasurements;
```
Comment (ProfFan) [inline gtsam/navigation/AHRSFactor.h:201]: I think `GTSAM_DEPRECATED` can be used, but not very sure if that would compile

**Review by ProfFan — APPROVED** [review]

File: gtsam/3rdparty/Eigen/Eigen/src/Core/CwiseNullaryOp.h [hunk 8]
```diff
@@ -215,7 +215,7 @@ DenseBase<Derived>::Constant(const Scalar& value)
   return DenseBase<Derived>::NullaryExpr(RowsAtCompileTime, ColsAtCompileTime, internal::scalar_constant_op<Scalar>(value));
 }
 
-/** \deprecated because of accuracy loss. In Eigen 3.3, it is an alias for LinSpaced(Index,const Scalar&,const Scalar&)
+/** @deprecated because of accuracy loss. In Eigen 3.3, it is an alias for LinSpaced(Index,const Scalar&,const Scalar&)
```
[the remark this answers is not in the document]
Comment (dellaert) [reply gtsam/3rdparty/Eigen/Eigen/src/Core/CwiseNullaryOp.h:218]: Good catch! Reverting!

File: gtsam/geometry/Cal3.h (hunk 3 above)
Comment (dellaert) [reply gtsam/geometry/Cal3.h:175]: Done

File: gtsam/geometry/Cal3Bundler.h (hunk 4 above)
Comment (dellaert) [reply gtsam/geometry/Cal3Bundler.h:102]: done

File: gtsam/linear/GaussianConditional.cpp (hunk 6 above)
Comment (dellaert) [reply gtsam/linear/GaussianConditional.cpp:197]: Done

File: gtsam/navigation/AHRSFactor.h (hunk 7 above)
Comment (dellaert) [reply gtsam/navigation/AHRSFactor.h:201]: Leaving as is

**Comment by dellaert** [comment]
Wow, thanks for catching 3dparty changes. That was a blunder :-)

**Pull request merged**